Sources/TensorFlow/Core/LazyTensorShapeInference.swift

```diff
@@ -66,6 +66,38 @@ extension LazyTensorOperation {
             }
         }
 
+        // Return the CTensor, selectively materializing it if needed.
+        func cTensor(handle: LazyTensorHandle) -> CTensor? {
+            switch handle.handle {
+            case .concrete(let h, _):
+                let cTensor = TFE_TensorHandleResolve(h._cTensorHandle, status)
+                checkOk(status)
+                return cTensor
+            case .symbolic(let op, _, _):
+                // TODO(https://bugs.swift.org/browse/TF-765): "Pack" is used
+                // for creating tensors from array literals. So, allow
+                // materialization for 'Pack' so that we can get the shape for
+                // array literals. We should revisit this heuristic.
+                if op.name != "Pack" { return nil }
+                let cTensor = TFE_TensorHandleResolve(handle._cTensorHandle, status)
+                checkOk(status)
+                return cTensor
+            }
+        }
+
+        // Create input_tensors consisting of *only* materialized inputs.
+        var input_tensors = Array<CTensor?>()
```

Comment:
* Camel case.
* Always use the sugared type expression for `Array`, `Dictionary`, and `Optional` types when a generic argument exists.
* Prefer array literal initialization.
```suggestion
        var inputTensors: [CTensor?] = []
```